### PR TITLE
Evaluation Queue Page

### PR DESCRIPTION
**Pull request by thomasyu888**

### CODE REVIEW

**Comment by thomasyu888** [comment]
Should I leave a description of all the columns that could be added to the Leaderboard?

Furthermore, should `Submissions` be a page on its own?  

There are functions that interact with submissions and the submission status annotations are also linked to what can be displayed on a leaderboard.  Should this page be called Evaluations and Submissions?

File: articles/evaluation_queues.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: Evaluation Queues
+layout: article
+excerpt: Evaluations queues are designed to support open-access data analysis and modeling challenges in Synapse.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: not a definition of an evaluation queue

File: articles/evaluation_queues.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+---
+title: Evaluation Queues
+layout: article
+excerpt: Evaluations queues are designed to support open-access data analysis and modeling challenges in Synapse. 
+category: howto
+---
+
+# Evaluation Queues
+Evaluations queues are designed to support open-access data analysis and modeling challenges in Synapse. This framework provides tools for administrators to collect and analyze data models from Synapse users created for a specific goal or purpose.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: same as above - need a definition of what a queue is, not just what it's for

File: articles/evaluation_queues.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+## Create an Evaluation Queue 
+
+Synapse users can submit Synapse Files, or Synapse Docker containers to an Evaluation queue ("Evaluation" for short) which you manage.  An Evaluation has a "contentSource" field which during creation should be set the Synapse ID of your challenge project.  Afterwards your project will have a **Challenge** tab next to the Wiki, Files and Tables tabs.  Your new Evaluation(s) will appear in a list.   You can create a new Evaluation using the R, Python or web client.  There are also ways to define submission 'rounds' (start date, round duration, and number of rounds) with optional submission quota (maximum submissions per participant or team) for each Evaluation queue.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: can submit any Synapse entity to a queue

Comment (kdaily) [reply articles/evaluation_queues.md]: seems backwards to suggest to create an evaluation queue before having a challenge set up.

Comment (kdaily) [reply articles/evaluation_queues.md]: can you have a queue without a challenge?

Comment (kdaily) [reply articles/evaluation_queues.md]: a lot of content that describes what happens after creating - should be after the steps that actually do it (the code block below)

Comment (kdaily) [reply articles/evaluation_queues.md]: No need to call out all clients here, clear from context below (might only mention if one of the clients didn't support it)

File: articles/evaluation_queues.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+You can create evaluation queues on the web by navigating to your challenge site by adding `/admin` to the url (E.g. www.synapse.org/#!Synapse:syn12345/admin).  Click **Tools** on the right corner and **Add Evaluation Queue** and follow the instructions on the screen.
+
+<img src="/assets/images/create_evaluation_queues.png">
+
+	{% endtab %}
+{% endtabs %}
+
+## Configure an Evaluation Queue 
+
+Unforuntately there is not a way to configure the round or quota settings of an evaluation queue from the web. The evaluation id can be found under the **Challenge** tab.  In the case below, the evaluation queue id is 9610091.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: Remove 'unfortunately'.

'There is currently no way to configure....'

File: articles/evaluation_queues.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+		numberOfRounds=1), #	The number of rounds, or null if there is no end. (Based on the duration of each round)
+	contentSource="syn12345", #Your synapse project id
+	submissionInstructionsMessage="Instructions on submission format...",
+	submissionReceiptMessage="Thanks for submitting to My Example Challenge!")
+synStore(evaluation)
+		{%endhighlight %}
+	{% endtab %}
+
+	{% tab Web %}
+You can create evaluation queues on the web by navigating to your challenge site by adding `/admin` to the url (E.g. www.synapse.org/#!Synapse:syn12345/admin).  Click **Tools** on the right corner and **Add Evaluation Queue** and follow the instructions on the screen.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: Evaluation

Comment (kdaily) [reply articles/evaluation_queues.md]: all code should use `synapser`

File: articles/evaluation_queues.md (hunk 3 above)
Comment (kdaily) [reply articles/evaluation_queues.md]: too many details about rounds and configuration here.

File: articles/evaluation_queues.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+synStore(evaluation)
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
+
+
+## Share an Evaluation Queue
+
+Each Evaluation has its own sharing settings.  The sharing levels are: "Administrator", "Can score", "Can Submit", and "Can view."
```
Comment (kdaily) [inline articles/evaluation_queues.md]: redundant with the info below

File: articles/evaluation_queues.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+evalId = 9610091
+evaluation <- synGetEvaluation(evalId)
+evaluation$quota = synapseClient:::SubmissionQuota(firstRoundStart = '2017-11-02T07:00:00.000Z', #The date/time at which the first round begins in UTC.
+	roundDurationMillis = 1645199000, #The duration of each round.
+	numberOfRounds=1) #	The number of rounds, or null if there is no end. (Based on the duration of each round) 
+synStore(evaluation)
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
```
Comment (kdaily) [inline articles/evaluation_queues.md]: Python and R examples should be identical.

File: articles/evaluation_queues.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+
+## Share an Evaluation Queue
+
+Each Evaluation has its own sharing settings.  The sharing levels are: "Administrator", "Can score", "Can Submit", and "Can view."  
+- "Administrator" sharing should be tightly restricted, as it includes authority to delete the entire Evaluation queue with all its contents. These users also have the ability to download all the submissions.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: They can download even in the presence of sharing settings or access restrictions that may not allow?

File: articles/evaluation_queues.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+- "Can score" allows for individuals to download all the submissions
+- "Can submit" allows for Teams or individuals to submit to the Evaluation, but doesn't have access to any of the submissions.
+- "Can view" allows for Teams or individuals to view the submissions on a leaderboard.
+
+To set the sharing setting, go to the **Challenge** tab and see your list of Evaluations.  Click on the `Share` button per Evaluation and share it with the Teams or individuals you would like.
+
+
+## View Submissions of an Evaluation Queue
+
+All submissions of an evaluation queue can be views through the use of leaderboard widgets in Wiki pages.  To learn how to create a wiki page, please visit [here](http://docs.synapse.org/articles/wikis.html).  Below are instructions on how to set up a leaderboard. You must know the evaluation id to do so.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: Evaluation queue

Comment (kdaily) [inline articles/evaluation_queues.md]: through the use of a leaderboard. you use a leaderboard widget in a wiki.

File: articles/evaluation_queues.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+#### Adding Leaderboard Widget
+
+<img style="width: 80%;" src="/assets/images/add_leaderboard_widget.png">
+
+#### Configuring Leaderboard Widget
+
+Once you click on **Leaderboard**, you will have to input your own query statement such as `select * from evaluation_9610091`.  To view all the columns available, click **Refresh Columns**.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: what's a query statement and where does the ID come from

**Review by kdaily — CHANGES_REQUESTED** [review]
need to rewrite R examples with `synapser`.

File: articles/evaluation_queues.md (hunk 3 above)
Comment (thomasyu888) [reply articles/evaluation_queues.md]: You can physically have a queue without a challenge, but I'll have to check if you can submit to it at all without having the project being a "challenge"

File: articles/evaluation_queues.md (hunk 8 above)
Comment (thomasyu888) [reply articles/evaluation_queues.md]: Submission objects are weird.  All file entities submitted are automatically "duplicated" in synapse.  So yes, all the submissions are always available.

Comment (kdaily) [reply articles/evaluation_queues.md]: I understand why - is that documented explicitly anywhere? We should be clear that sharing settings go out the window when submitting to a Evaluation queue.

Comment (thomasyu888) [reply articles/evaluation_queues.md]: I added a statement about this.

**Comment by ychae** [comment]
Bruce mentioned in a Jira that:
```
The 'status' of an Evaluation is deprecated / no longer used. 
It has been replaced by the Evaluation ACL. Perhaps it's time to remove it from our API altogether.
```

Should we modify the example to reflect that and remove `status` altogether?

**Comment by kdaily** [comment]
Not sure how I understand how the ACL and the Evaluation status are related? So, defer to @thomasyu888 for that.

**Comment by thomasyu888** [comment]
I don't think they are related.  The ACL of an evaluation queue basically allows people to view public leaderboards, admins download submissions and such.  I have never used the status of an Eval queue before.

File: articles/evaluation_queues.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+You can create Evaluation queues on the web by navigating to your challenge site by adding `/admin` to the url (E.g. www.synapse.org/#!Synapse:syn12345/admin).  Click **Tools** on the right corner and **Add Evaluation Queue** and follow the prompts.
+
+<img src="/assets/images/create_evaluation_queues.png">
+
+{% endtab %}
+{% endtabs %}
+
+## Configure an Evaluation Queue 
+
+Submission "rounds" (start date, round duration, and number of rounds) with an optional submission quota (maximum submissions per participant or team) can be defined for each queue.  There is not a way to configure the round or quota settings of an evaluation queue from the web. The Evaluation ID can be found under the **Challenge** tab of your project. Please note that a Challenge tab will not appear on your project until you have created a challenge (**Tools->Run Challenge**). In the case below, the evaluation queue id is `9610091`.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: sentence is confusing - separate concepts of rounds from quotas.

Comment (kdaily) [reply articles/evaluation_queues.md]: "There is no way to configure..."

Comment (kdaily) [reply articles/evaluation_queues.md]: "...settings of an Evaluation queue..."

File: articles/evaluation_queues.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+{%endhighlight %}
+{% endtab %}
+
+{% endtabs %}
+
+
+## Share an Evaluation Queue
+
+Each Evaluation has its own sharing settings.  The sharing levels are: "Administrator", "Can score", "Can Submit", and "Can view."  
+- "Administrator" sharing should be tightly restricted, as it includes authority to delete the entire Evaluation queue with all its contents. These users also have the ability to download all the submissions.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: "all submissions"

File: articles/evaluation_queues.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+
+To set the sharing setting, go to the **Challenge** tab and see your list of Evaluations.  Click on the `Share` button per Evaluation and share it with the Teams or individuals you would like.
+
+
+{% include important.html content="When someone submits to an Evaluation, a copy of the submission is made, so a person with Administrator or Can score access will be able to download the submission even if the submitter deletes the entity." %}
+
+
+## View Submissions of an Evaluation Queue
+
+All submissions of an Evaluation queue can be views through the through the use of a leaderboard.  To learn how to create a wiki page, please visit [here](http://docs.synapse.org/articles/wikis.html).  Below are instructions on how to set up a leaderboard. You must know the **evaluation Id** to do so.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: typos/duplicate text

Comment (kdaily) [reply articles/evaluation_queues.md]: how do i learn the evaluation id? point to another section in this doc.

File: articles/evaluation_queues.md [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+#### Adding Leaderboard Widget
+
+<img style="width: 80%;" src="/assets/images/add_leaderboard_widget.png">
+
+#### Configuring Leaderboard Widget
+
+Once you click on **Leaderboard**, you will have to input your own query statement such as `select * from evaluation_9610091`.  Remember, 9610091 should be replaced with your own personal evaluation Id. To view all the columns available, click **Refresh Columns**.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: not a personal evaluation ID, just your own.

File: articles/evaluation_queues.md [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+
+<img style="width: 80%;" src="/assets/images/configure_leaderboard_widget.png">
+
+Clicking **Refresh Columns** will add these default columns.
+
+<img style="width: 80%;" src="/assets/images/leaderboard_columns.png">
+
+#### Saving Leaderboard Widget
+If you are happy with your leaderboard configurations, save both the configurations and the wiki page and you will see something like this.
```
Comment (kdaily) [inline articles/evaluation_queues.md]: remove "see something like this".

**Review by kdaily — CHANGES_REQUESTED** [review]

File: articles/evaluation_queues.md (hunk 11 above)
Comment (meredithslota) [reply articles/evaluation_queues.md]: Fixed.

File: articles/evaluation_queues.md (hunk 6 above)
Comment (meredithslota) [reply articles/evaluation_queues.md]: Fixed.

File: articles/evaluation_queues.md (hunk 14 above)
Comment (meredithslota) [reply articles/evaluation_queues.md]: Fixed.

File: articles/evaluation_queues.md (hunk 13 above)
Comment (meredithslota) [reply articles/evaluation_queues.md]: Fixed.

File: articles/evaluation_queues.md (hunk 15 above)
Comment (meredithslota) [reply articles/evaluation_queues.md]: Fixed.

**Review by meredithslota — APPROVED** [review]
Made the changes that @kdaily requested and approving.

**Pull request merged**